### PR TITLE
Fix for installation of exported targets

### PR DESCRIPTION
**Pull request by Yangqing**
cc @houseroad @pietern

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -139,27 +141,38 @@ endif()
 
 # ---[ Make configuration files for cmake to allow dependent libraries
 # easier access to Caffe2.
+# Since one would use glog, gflags and protobuf functionalities, any
```
Comment (pietern) [inline CMakeLists.txt]: oxford comma

File: CMakeLists.txt [hunk 2]
```diff
@@ -139,27 +141,38 @@ endif()
...
+if (USE_GLOG)
+  list(APPEND CAFFE2_INTERFACE_LIBS glog)
+endif()
+if (USE_GFLAGS)
+  list(APPEND CAFFE2_INTERFACE_LIBS gflags)
+endif()
+if (NOT CAFFE2_USE_CUSTOM_PROTOBUF)
+  list(APPEND CAFFE2_INTERFACE_LIBS protobuf)
+endif()
+# message(STATUS "Caffe2 interface libs: " ${CAFFE2_INTERFACE_LIBS})
```
Comment (pietern) [inline CMakeLists.txt]: comment

**Comment by pietern** [comment]
@Yangqing Nice. The only difference is that for BUILD_SHARED=OFF you now don't declare the generate CMake files right? The CAFFE2_INTERFACE_LIBS stuff is sugar?

**Comment by Yangqing** [comment]
@pietern Yep. Basically for OSS I think we should use shared library as the standard deployment path, as static library + imported dependency is literally something unmanageable :)

For Android use cases, this means that users have to hand pick what static libs they want to link against, but that is probably the best because it consciously makes users aware of binary size.

File: CMakeLists.txt (hunk 1 above)
Comment (Yangqing) [reply CMakeLists.txt]: done

File: CMakeLists.txt (hunk 2 above)
Comment (Yangqing) [reply CMakeLists.txt]: done

**Review by pietern — APPROVED** [review]
There's the cmake policy at the start, do you want to keep those around?

**Comment by Yangqing** [comment]
Yep, the policy is there so we don't accidentally mix old style and new style target_link_libraries.

**Pull request closed without merging**